util/init_context.go

```diff
@@ -46,7 +56,8 @@ func InitAmfContext(context *context.AMFContext) {
 	if configuration.NrfUri != "" {
 		context.NrfUri = configuration.NrfUri
 	} else {
-		context.NrfUri = fmt.Sprintf("%s://%s:%d", context.UriScheme, context.HttpIPv4Address, 29510)
+		logger.UtilLog.Warn("NRF Uri is empty! Using localhost as NRF IPv4 address.")
+		context.NrfUri = fmt.Sprintf("%s://%s:%d", context.UriScheme, "127.0.0.1", 29510)
```

Comment:
We except if the nrf not write, NRF will be at the same IP (same PC) as NF (amf here).

Or if there any scenario to guest localhost will be better?

Reply:
Well I think in the end it's the same thing actually.
So let's say there are two scenarios possible:
1. __All NFs are placed together in the same host__: Then basically it doesn't matter if `localhost`  or the same IP as the NF is used.
2. __NF are placed in at least two different nodes__: In that case, it again does not make any difference. If we consider a scenario where for e.g. AMF and NRF are placed in __Host_1__ and SMF is placed in __Host_2__, and the `NrfUri` is empty in both config files of AMF and SMF, we will have the following cases:
  a) If we are using SameIP, AMF will manage to successfully connect to NRF, while SMF will fail since NRF is running on another node (they don't have the same IP).
  b) If we are using `localhost`, AMF will manage to connect to NRF while SMF will fail.

In such case, I though it is better to use the `localhost` as the default since it is used more generally as the fallback IP.

However, what I wanted to emphasize with this changes was that the NF logs a Warning in case the `NrfUri` is not configured, and the log gives some information on what is the fallback IP that is used.

Reply:
You are right. This sounds reasonable.